src/Auth/SignUp.js

```diff
@@ -0,0 +1,195 @@
+            onChange={(e) =>
+              setSignUpCredentials(() => ({
+                ...signUpCredentials,
+                msg: "",
+                username: e.target.value,
+              }))
+            }
+          />
+          <span className='flt-label flt-label-form tri-pink'>
+            <i className='fas fa-user'></i> Username
+          </span>
+        </div>
+
+        <div className='login-input'>
+          <input
+            className='input-txt-error'
+            required
+            type={signUpCredentials.showPassword ? "text" : "password"}
+            value={signUpCredentials.password}
+            onChange={(e) =>
+              setSignUpCredentials(() => ({
+                ...signUpCredentials,
+                msg: "",
+                password: e.target.value,
+              }))
+            }
+          />
+          <span className='flt-label flt-label-form tri-pink'>
+            <i className='fas fa-lock'></i> Password
+          </span>
+          <button
+            className='show-pass'
+            onClick={() =>
+              setSignUpCredentials(() => ({
+                ...signUpCredentials,
+                showPassword: !signUpCredentials.showPassword,
+              }))
+            }
+          >
+            {signUpCredentials.showPassword ? (
+              <i className='far fa-lg fa-eye-slash'></i>
+            ) : (
+              <i className='far fa-lg  fa-eye'></i>
+            )}
+          </button>
+        </div>
+
+        <div className='login-input'>
+          <input
+            className='input-txt-error'
+            required
+            type={signUpCredentials.showConfirmPassword ? "text" : "password"}
+            value={signUpCredentials.confirmPassword}
+            onChange={(e) =>
+              setSignUpCredentials(() => ({
+                ...signUpCredentials,
+                msg: "",
+                confirmPassword: e.target.value,
+              }))
+            }
+          />
+          <span className='flt-label flt-label-form tri-pink'>
+            <i className='fas fa-lock'></i> Confirm Password
+          </span>
+          <button
+            className='show-pass'
+            onClick={() =>
+              setSignUpCredentials(() => ({
+                ...signUpCredentials,
+                showConfirmPassword: !signUpCredentials.showConfirmPassword,
+              }))
+            }
+          >
+            {signUpCredentials.showConfirmPassword ? (
+              <i className='far fa-lg fa-eye-slash'></i>
+            ) : (
+              <i className='far fa-lg  fa-eye'></i>
+            )}
+          </button>
+        </div>
+
+        <p>{signUpCredentials.msg}</p>
+
+        <p className='mg'>
+          Already have an account?
+          <Link to='/login'>
+            <span className='pink-txt'>Login up!</span>
+          </Link>
+        </p>
+
+        <button className='btn btn-main' onClick={signUpHandler}>
```

Comment:
SignUp or SignUpNewUser is itself self explanatory, so we should avoid "handler" in function names